File: 2358/2358.cs
public class Solution {
    public int MaximumNumberOfGroupsEnteringA(TreeNode root) {
        if (root == null) return 0;
        return 1 + Math.Max(MaximumNumberOfGroupsEnteringA(root.left), MaximumNumberOfGroupsEnteringA(root.right));
    }
}

File: 2358/2358.py
class Solution:
    def maximum_number_of_groups_entering_a_comp(self, root: Optional[TreeNode]) -> int:
        if not root:
            return 0
        return 1 + max(self.maximum_number_of_groups_entering_a_comp(root.left), self.maximum_number_of_groups_entering_a_comp(root.right))